llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -6081,12 +6081,12 @@ Instruction *InstCombinerImpl::foldICmpWithCastOp(ICmpInst &ICmp) {
 
   // Turn icmp (ptrtoint x), (ptrtoint/c) into a compare of the input if the
   // integer type is the same size as the pointer type.
-  auto CompatibleSizes = [&](Type *SrcTy, Type *DestTy) {
-    if (isa<VectorType>(SrcTy)) {
-      SrcTy = cast<VectorType>(SrcTy)->getElementType();
-      DestTy = cast<VectorType>(DestTy)->getElementType();
+  auto CompatibleSizes = [&](Type *PtrTy, Type *IntTy) {
+    if (isa<VectorType>(PtrTy)) {
+      PtrTy = cast<VectorType>(PtrTy)->getElementType();
+      IntTy = cast<VectorType>(IntTy)->getElementType();
     }
-    return DL.getPointerTypeSizeInBits(SrcTy) == DestTy->getIntegerBitWidth();
+    return DL.getPointerTypeSizeInBits(PtrTy) == IntTy->getIntegerBitWidth();
```

Comment:
Why the variable rename?

Reply:
The existing code for `ptrtoint` calls `CompatibleSizes(SrcTy, DestTy)`. The code that I added for `inttoptr` calls with the arguments in the reverse order: `CompatibleSizes(DestTy, SrcTy)`. It felt awkward to pass a `Dest` to a function expecting `Src` and vice versa. I figured that anyway the more important distinction is which is the int and which is the pointer.